servicetest/networkgateway/test_network_gateway.py

```diff
@@ -0,0 +1,233 @@
+# any purpose with or without fee is hereby granted, provided that the
+# above copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+
+import pytest
+
+import os
+import time
+
+import testhelper
+from testframework import Container
+
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+# This function is used by the 'agent' fixture to know where the log should be stored
+def logfile_path():
+    return CURRENT_DIR + "/test.log"
+
+
+# These default values are used to pass various test specific values and
+# configurations to the Container helper. Tests that need to add, remove or
+# update entries can simply base their dict on this one for convenience.
+DATA = {
+    Container.CONFIG: '[{"enableWriteBuffer": false}]',
+    Container.BIND_MOUNT_DIR: "app",
+    Container.HOST_PATH: CURRENT_DIR,
+    Container.READONLY: False
+}
+
+GW_CONFIG_POLICY_DROP = [
+        {
+            "type": "OUTGOING",
+            "priority": 1,
+            "rules": [],
+            "default": "DROP"
+        }
+]
+
+GW_CONFIG_POLICY_ACCEPT = [
+        {            
+            "type": "OUTGOING",
+            "priority": 1,
+            "rules": [],
+            "default": "ACCEPT"
+        }
+]
+
+GW_CONFIG_REJECT_EXAMPLE_COM = [
+        {            
+            "type": "OUTGOING",
+            "priority": 1,
+            "rules": [{"host": "example.com", "target": "REJECT"}],
+            "default": "ACCEPT"
+        }
+]
+
+GW_CONFIG_REJECT_EXAMPLE_COM_INPUT = [
+        {            
+            "type": "INCOMING",
+            "priority": 1,
+            "rules": [{"host": "example.com", "target": "REJECT"}],
+            "default": "ACCEPT"
+        }
+]
+
+
+
+@pytest.mark.usefixtures("agent")
+class TestNetworkRules(object):
+    """ This suite tests that whether NetworkGateway can parse given configuration and applies it to be used 
+        in SoftwareContainer with expected results.
+
+        timeouts are 
+            2 seconds for expected connectivity
+            10 for expected output discconnectivity
+            15 for expected input connectivity
+
+        these durations will be shorten with -i option when possible    
+    """
+    def test_network_policy_drop(self):
+        """
+            Test setting network default target to drop and tests the corresponding behavior
+            which is being not possible to ping a hostname
+        """
```

Comment:
In all of these, pull up the first text line so it's part of the opening quotes, like you've done in the class comment above.